### PR TITLE
overlord: switch snapstate.Update to use ListRefresh (aka /snaps/metadata)

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: tests/lib/fakestore/store/store.go [hunk 1]
```diff
@@ -252,6 +252,7 @@ var snapIDtoName = map[string]string{
 	"b8X2psL1ryVrPt5WEmpYiqfr5emixTd7": "ubuntu-core",
 	"bul8uZn9U3Ll4ke6BMqvNVEZjuJCSQvO": "canonical-pc",
 	"SkKeDk2PRgBrX89DdgULk3pyY5DJo6Jk": "canonical-pc-linux",
+	"eFe8BTR5L5V9F7yHeMAPxkEr2NdUXMtw": "test-snapd-tools",
```
Comment (mvo5) [inline tests/lib/fakestore/store/store.go]: Nice!

File: overlord/managers_test.go [hunk 2]
```diff
@@ -225,14 +225,27 @@ apps:
...
+
 	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.URL.Path {
 		case "/details/foo":
 			w.WriteHeader(http.StatusOK)
-			output := strings.Replace(fooSearchHit, "@URL@", baseURL+"/snap", -1)
-			output = strings.Replace(output, "@ICON@", baseURL+"/icon", -1)
-			output = strings.Replace(output, "@VERSION@", ver, -1)
-			output = strings.Replace(output, "@REVISION@", revno, -1)
+			io.WriteString(w, fillHit())
```
Comment (mvo5) [inline overlord/managers_test.go]: This looks a bit unreleated? Does not hurt of course, just wondering

Comment (mvo5) [inline overlord/managers_test.go]: Oh, nevermind, sorry

**Comment by mvo5** [comment]
:+1:

**Comment by chipaca** [comment]
:+1:

**Pull request merged**